pkg/providers/amifamily/ami.go

```diff
@@ -94,45 +98,91 @@ func (p *Provider) KubeServerVersion(ctx context.Context) (string, error) {
 	return version, nil
 }
 
-// Get returns a set of AMIIDs and corresponding instance types. AMI may vary due to architecture, accelerator, etc
+// MapInstanceTypes returns a set of AMIIDs and corresponding instance types. AMI may vary due to architecture, accelerator, etc
 // If AMI overrides are specified in the AWSNodeTemplate, then only those AMIs will be chosen.
-func (p *Provider) Get(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate, instanceTypes []*cloudprovider.InstanceType, amiFamily AMIFamily) (map[string][]*cloudprovider.InstanceType, error) {
-	kubernetesVersion, err := p.KubeServerVersion(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("getting kubernetes version %w", err)
-	}
+func MapInstanceTypes(amis []AMI, instanceTypes []*cloudprovider.InstanceType) map[string][]*cloudprovider.InstanceType {
 	amiIDs := map[string][]*cloudprovider.InstanceType{}
-	amiRequirements, err := p.getAMIRequirements(ctx, nodeTemplate)
-	if err != nil {
-		return nil, err
-	}
-	if len(amiRequirements) > 0 {
-		// Iterate through AMIs in order of creation date to use latest AMI
-		amis := sortAMIsByCreationDate(amiRequirements)
-		for _, instanceType := range instanceTypes {
-			for _, ami := range amis {
-				if err := instanceType.Requirements.Compatible(amiRequirements[ami]); err == nil {
+
+	for _, instanceType := range instanceTypes {
+		for _, ami := range amis {
+			if !resources.IsZero(instanceType.Capacity[v1alpha1.ResourceAWSNeuron]) {
```

Comment:
This seems like a change to the core AMI discovery logic. IIUC, this is now considering instance types that have AWSNeuron resources, and injecting a new requirement. 

Even if it's needed for this PR, it seems like scope creep. Is it possible to add this in a different PR?

Reply:
I was wondering if this isn't already a requirement on the instance type to begin with. Ideally, we should be able to understand that the instance type matches the scheduling requirements without even touching the capacity resources